python/ray/experimental/data/impl/arrow_block.py

```diff
@@ -171,3 +176,41 @@ def size_bytes(self) -> int:
     @staticmethod
     def builder() -> ArrowBlockBuilder[T]:
         return ArrowBlockBuilder()
+
+    def sample(self, n_samples: int, key: SortKeyT) -> List[T]:
+        k = min(n_samples, self._table.num_rows)
+        indices = random.sample(range(self._table.num_rows), k)
+        return self._table.select([k[0] for k in key]).take(indices)
+
+    def sort_and_partition(self, boundaries: List[T], key: SortKeyT,
+                           descending: bool) -> List["Block[T]"]:
+        if len(key) > 1:
+            raise NotImplementedError(
+                "sorting by multiple columns is not supported yet")
+
+        import pyarrow.compute as pac
+
+        indices = pac.sort_indices(self._table, sort_keys=key)
+        table = self._table.take(indices)
+        if len(boundaries) == 0:
+            return [table]
+        col, _ = key[0]
+        comp_fn = pac.greater if descending else pac.less
+        boundary_indices = [
+            pac.sum(comp_fn(table[col], b)).as_py() for b in boundaries
+        ]
+        ret = []
+        prev_i = 0
+        for i in boundary_indices:
+            ret.append(table.slice(prev_i, i - prev_i))
+            prev_i = i
+        ret.append(table.slice(prev_i))
+        return ret
+
+    @staticmethod
+    def merge_sorted_blocks(blocks: List[Block[T]], key: SortKeyT,
+                            _descending: bool) -> Block[T]:
```

Comment:
Is the return type correct? Shouldn't it return a tuple?

Reply:
will fix